app/app.go

```diff
@@ -76,6 +78,8 @@ func appHandler(c *echo.Context) error {
 				log.Error(err)
 				failed = true
 			}
+			appsMutex.Lock()
+			defer appsMutex.Unlock()
```

Comment:
I tend to do all `Unlock()`s in `defer` statements, but this is very simple and really should only fail here on a concurrent map write, which this change fixes anyway